Update readme to better reflect what we have today (#508)
cert-manager/trust-manager: README.md

trust-manager is the easiest way to manage trust bundles in Kubernetes and OpenShift clusters!

> It takes a list of trusted certificate sources and combines them into a bundle which you can trust directly

Author: I think we could go into more detail here but I think we'd be duplicating the docs. I'd rather not change this _too_ much, and instead direct people to read the website.

My feeling is that it'd be best to leave this as-is, and maybe revisit later. I'd prefer to quickly improve this and move on for now if that's OK?

> It takes a list of trusted certificate sources and combines them into a bundle which you can trust directly

Reviewer: Sounds good!

- [Installation instructions](https://cert-manager.io/docs/trust/trust-manager/installation/)
- [Usage guidance](https://cert-manager.io/docs/trust/trust-manager/)
- [API reference documentation](https://cert-manager.io/docs/trust/trust-manager/api-reference/)

## Developing trust-manager

trust-manager uses [makefile-modules](https://github.com/cert-manager/makefile-modules/), meaning that any changes to files under `make/_shared` need to be made in that repo and synchronized here using `make upgrade-klone`.

The easiest way to get started is to run the trust-manager smoke tests locally.

Use `make test-smoke`, which creates a [Kind cluster](https://kind.sigs.k8s.io/) using Docker and installs trust-manager (and cert-manager) before running the tests.

To create a cluster without running the smoke tests, use `make test-smoke-deps`.

To use or inspect the cluster, the `KUBECONFIG` file needs to be made available:

```console
export KUBECONFIG=$(pwd)/_bin/scratch/kube.config
```

### Testing

trust-manager has various categories of tests. All categories are run against every PR, along with other checks.

- `make test-unit` - Runs simpler, faster tests which test specific functions

> - `make test-integration` - Runs heavier tests with no Kubernetes cluster which test how different pieces work together

Reviewer: ```suggestion
- `make test-integration` - Runs heavier tests with a simplified control-plane which tests how different pieces work together
```
